clippy_lints/src/utils/mod.rs

```diff
@@ -50,16 +50,34 @@ use rustc_lint::{LateContext, Level, Lint, LintContext};
 use rustc_middle::hir::map::Map;
 use rustc_middle::ty::subst::{GenericArg, GenericArgKind};
 use rustc_middle::ty::{self, layout::IntegerExt, Ty, TyCtxt, TypeFoldable};
+use rustc_session::Session;
 use rustc_span::hygiene::{ExpnKind, MacroKind};
 use rustc_span::source_map::original_sp;
 use rustc_span::symbol::{self, kw, Symbol};
 use rustc_span::{BytePos, Pos, Span, DUMMY_SP};
 use rustc_target::abi::Integer;
 use rustc_trait_selection::traits::query::normalize::AtExt;
+use semver::Version;
 use smallvec::SmallVec;
 
 use crate::consts::{constant, Constant};
 
+
+pub fn check_msrv(sess: &Session, msrv: &String) {
```

Comment:
There is no good place, where I can leave this comment, so I just write it here.

Why not turn this function into `parse_msrv(..) -> Option<Version>`?

Then we can use it in `clippy_lints/src/utils/conf.rs` and in the `enter_lint_attrs` methods. 

With that we also won't have to parse the `msrv` with each call to `meets_msrv`, but just save a `semver::Version` in the `LintPass` structs.

In addition to that, we should pass a `Option<Span>` to this function, that is `None` in `clippy_lints/src/utils/conf.rs`, but the span of the attribute in `enter_lint_attrs` to improve the error message.

---

I hope, I expressed my plan clear enough. If not, just ask if anything is unclear.

Reply:
@flip1995 Thank you for reviewing! If I understand correctly, we need the following changes 

- update all the lints to store the `msrv` as a `semver::Version` instead of a `String` so that we don't have to repeatedly parse it while checking
- convert `check_msrv(..)` to `parse_msrv(..) -> Option<Version>`. We throw an error if `parse_msrv(..)` returns `None`

I think these make sense, I'm just wondering how to pass the `Span` to `parse_msrv(..)` since we don't have a handle on the `Attribute`. Do you think we should change the signature of `get_inner_attr` to accept an `FnMut(Attribute)`? or should it just return the attribute instead of passing it to a closure

Reply:
I would let `get_inner_attr` return `(String, Span)`. Or alternatively pass the span in the closure. I don't really mind what you choose here. Personally I like to avoid closures, if they aren't necessary (because they can be confusing for new programmers).